runtime_scan/pkg/provider/aws/client.go

```diff
@@ -37,12 +37,21 @@ import (
 
 type Client struct {
 	ec2Client *ec2.Client
-	awsConfig *Config
+	config    *Config
 }
 
-func New(ctx context.Context, config *Config) (*Client, error) {
+func New(ctx context.Context) (*Client, error) {
+	config, err := NewConfig()
+	if err != nil {
+		return nil, fmt.Errorf("invalid configuration. Provider=AWS: %w", err)
+	}
+
+	if err = config.Validate(); err != nil {
+		return nil, fmt.Errorf("failed to validate provider configuration. Provider=AWS: %w", err)
+	}
```

Comment:
Is there much point in having this separate from NewConfig()?

Reply:
The reason for having validation not being part of the `NewConfig`, because there might be cases where:
* the Config is updated after creating it with `NewConfig` therefore we do not want to fail the config creation
* the caller of `NewConfig` does custom validation